lib/Runtime/HostManager/HostManager.cpp

```diff
@@ -228,6 +228,11 @@ Error HostManager::addNetwork(std::unique_ptr<Module> module,
     return err;
   }
 
+  // Create pool of cachedExecutionStates.
+  for (auto &node : nodeList) {
```

Comment:
This potentially wastes quite a bit memory. Let's keep it this way and see whether the trade-off is worth it.

Reply:
Actually do we need to set default as number of inflight requests? I'd think it should be number of workers that will feed up the backend? Say 2?

Reply:
I think it needs to be maxActiveRequests for now. Since executor->run() modifies the networkExecutionState. say maxActive = 10 and workers = 2. If you have 10 requests, the first 2 would be enqueued and acted on immediately so we need at least 2. The next 8 would be dispatched to the executor, which would put them on the queue for the Device. Those 8 would need their networkExecutionState objects by then too.

Reply:
Yeah, it just doesn't necessarily need to be maxActiveRequests. One reason is that device resident tensors are going to consume resource in the cards. The other reason is that we can actually control the use of resource just by block waiting for the item to be available in the pool. Right now, let's put maxActiveRequests as an estimate. But we need to be aware that this number can change and system should still work.